app/scripts/lib/ComposableObservableStore.js

```diff
@@ -112,4 +120,29 @@ export default class ComposableObservableStore extends ObservableStore {
 
     this.emit('stateChange', { oldState, newState, controllerKey });
   }
+
+  /**
+   * Returns true if the given set of patches makes changes to a persisted property.
+   *
+   * Note that we assume at least one property is persisted, so a complete replacement patch
+   * always returns true.
+   *
+   * @param {StateMetadataConstraint} metadata - Controller metadata.
+   * @param {Patch[]} patches - A list of patches, corresponding to a single state update.
+   * @returns True if the patches contain a change to persisted state, false otherwise.
+   */
+  #changedPersistedProperty(metadata, patches) {
+    return patches.some((patch) => {
+      // Complete state replacement
+      if (patch.path.lenght === 0) {
+        return true;
+      }
+      const topLevelProperty = patch.path[0];
+      // Missing metadata, return true out of caution
+      if (!metadata[topLevelProperty]) {
+        return true;
+      }
```

Comment:
Is it correct to say that this branch would be reached only if a property was removed from a controller state without adding a state migration for it, but at the same time, only if the property was somehow mutated?

Reply:
Is there a test case for this scenario? I can't find one in the tests file - if not, do you think it would be worth adding one?